sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -1628,6 +1631,94 @@ def write_pipeline_spec_to_file(pipeline_spec: pipeline_spec_pb2.PipelineSpec,
         with open(package_path, 'w') as yaml_file:
             yaml.dump(json_dict, yaml_file, sort_keys=True)
 
+        with open(package_path, 'r+') as f:
+            old = f.read()
+            f.seek(0)
+            f.write(yaml_comments + old)
+
     else:
         raise ValueError(
             f'The output path {package_path} should end with ".yaml".')
+
+
+def extract_comments_from_pipeline_spec(pipeline_spec: dict,
+                                        pipeline_description: str) -> str:
+    map_parameter_types = {
+        'NUMBER_INTEGER': 'int',
+        'NUMBER_DOUBLE': 'float',
+        'STRING': 'str',
+        'BOOLEAN': 'bool',
+        'LIST': 'list',
+        'STRUCT': 'dict'
+    }
+
+    def add_inputs():
+        if 'inputDefinitions' in pipeline_spec['root']:
+            inputs = pipeline_spec['root']['inputDefinitions']
+            string = '# Inputs: \n'
+
+            if 'parameters' in inputs:
+                for parameter in inputs['parameters']:
+                    string += '#    ' + parameter + ': ' + map_parameter_types[
+                        inputs['parameters'][parameter]['parameterType']]
+                    if 'defaultValue' in inputs['parameters'][parameter]:
+                        string += ' [Default: ' + str(
+                            inputs['parameters'][parameter]
+                            ['defaultValue']) + ']'
+
+                    string += '\n'
+
+            if 'artifacts' in inputs:
+                for artifact in inputs['artifacts']:
+                    if 'schemaTitle' in inputs['artifacts'][artifact][
+                            'artifactType']:
+                        string += '#    ' + artifact + ': ' + inputs[
+                            'artifacts'][artifact]['artifactType'][
+                                'schemaTitle'] + '\n'
+                    #TODO: Add exception tp raise if schematitle doesnt exist
```

Comment:
I think this is a good idea. In practice, there should always be a schema title since we don't enable any other way of representing artifact schema in the KFP SDK, but it's good future-proof/defensive programming.

I think the `['schemaTitle']` indexing actually would have this effect in the current state, as a `KeyError` would be raised if the key is does not exist.

Perhaps we want to add this exception in explicitly, however, to make it clear that the exception is not coincidental. Up to you. Either way, I think it would be good to resolve this and remove the `TODO` before merging.

Reply:
I removed the ToDO and removed the if statement too, that way it raises the Key Error if `schemaTitle` doesn't exist.